force-app/main/default/classes/Shared Code/Log.cls

```diff
@@ -0,0 +1,119 @@
+/**
+ * @description Generic logging framework that persists across DML reversions
+ * by publishing a Platform Event
+ */
+public with sharing class Log {
```

Comment:
`inherited sharing` will make it flexible to handle security requirements when this class is called from the other classes.

Reply:
fixed!